Redirect after session expire

benjaminwols opened the pull request:
I made this PR to customise the redirect after session expire.
The default stays the same to invalid_session_path, but it is now possible to set your own route in the config.

Commit:
Add redirect_after_session expire method
Add redirect_after_session_expire method

Add default route session expire route in engine config

Add session expire redirect to config

Style update

Fix deprecation warning: replace validate_length_of with validate_length_of

Fix deprecation warning: replace serve_static_assets with serve_static_files

Fix depreaction warning: replace use_route with global declaration of routes

Add test for redirect_after_session_expire

Update readme for redirect_after_session_expired config setting

Revert "Fix deprecation warning: replace validate_length_of with validate_length_of"

This reverts commit 9ae0f92b1eb51b3632dcc4343faaac89f447aba9.

djcp commented:
Great! Can you look through `dce_lti/spec/controllers/dce_lti/sessions_controller_spec.rb` for examples  (and probably other spec files) and think of a test for this feature?

djcp commented on lib/dce_lti/controller_methods.rb:
```diff
@@ -1,9 +1,7 @@
 module DceLti
   module ControllerMethods
     def authenticate_via_lti
-      if ! current_user
```
I know this will seem anal, but I generally only use the `function unless foo` construction for guard clauses, and I don't really think of this as a guard clause. I could be convinced otherwise.

benjaminwols replied:
I reverted the change, but kept the if ! which I replaced with unless (which is the rails way https://github.com/bbatsov/ruby-style-guide#unless-for-negatives)

benjaminwols commented:
I will look at the test later this week.

benjaminwols commented:
I added a test for the redirect_after_session_expired. 
The test is based on 'redirects to "redirect_after_successful_auth" url when it is a proc' in sessions_controller_spec, but I put the test in posts_controller_spec, because I wanted to request posts index without a valid lti session, which than should redirect me to the url specified in the redirect_after_session_expire proc.

I fixed some deprecation warnings while adding a test.

Please let me know what you think of the test.

djcp commented:
This looks good - can you squash your commits, I'll give it one last look and merge? Thanks so much!

benjaminwols commented:
I squashed the commits.

djcp commented:
Thanks! I'm going to look at this (along with the newly activerecord_session-store release) and merge soon. You're a champ!
